pandas/io/parsers/base_parser.py

```diff
@@ -774,7 +774,10 @@ def _infer_types(
                     bool_mask = np.zeros(result.shape, dtype=np.bool_)
                 result = BooleanArray(result, bool_mask)
             elif result.dtype == np.object_ and use_nullable_dtypes:
-                result = StringDtype().construct_array_type()._from_sequence(values)
+                # read_excel sends array of datetime objects
+                inferred_type, _ = lib.infer_datetimelike_array(result)
+                if inferred_type != "datetime":
+                    result = StringDtype().construct_array_type()._from_sequence(values)
```

Comment:
Not too familiar with the parser code; we want to leave result as-is when it is datetime?

Reply:
Yeah, the DataFrame constructor is doing inference in this case